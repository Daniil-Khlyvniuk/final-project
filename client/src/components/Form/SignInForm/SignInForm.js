import React, { useState } from 'react'
import { Formik, Form, Field } from 'formik'
import { SING_UP_SCHEMA } from '../setting/Schemes'
import { useFormStyle } from '../../../utils/customHooks/useFormStyle'
import { Box, Typography, Button } from '@mui/material'
import { Link } from 'react-router-dom'
import CustomInput from '../setting/CustomInput'
import CustomCheckBox from '../setting/CustomCheckBox'
import CustomSwitch from '../setting/CustomSwitch'
import useAuth from '../../../utils/customHooks/useAuth'

const SignInForm = () => {
	const [serverResult, setServerResult] = useState(null)
	const { register } = useAuth()

	const classes = useFormStyle()

	return (
		<Box
			sx={{
				overflowY: 'auto',
				maxHeight: '530px',
				paddingRight: '10px'
			}}
		>
			<Formik
				initialValues={{
					firstName: '',
					lastName: '',
					login: '',
					email: '',
					password: '',
					confirmPass: '',
					subscribe: false,
					rememberMe: false
				}}
				validationSchema={SING_UP_SCHEMA}

				onSubmit={async (values) => {
					try {
						await register(values)
						setServerResult({ success: 'You successfully registered' })
					}
					catch (err) {
						setServerResult({ error: Object.values(err.response.data)[0] })
					}
				}}
			>
				{(formikProps) => {
					return (
						<Form noValidate
							onSubmit={formikProps.handleSubmit}
						>
							<Field
								component={CustomInput}
								data-testid="firstName"
								name="firstName"
								type="text"
								label="First Name"
							/>

							<Field
								component={CustomInput}
								data-testid="lastName"
								name="lastName"
								type="text"
								label="Last Name"
							/>

							<Field
								component={CustomInput}
								data-testid="login"
								name="login"
								type="text"
								label="Login"
								placeholder="Login"
							/>

							<Field
								component={CustomInput}
								data-testid="email"
								name="email"
								type="email"
								label="Email"
							/>

							<Field
								component={CustomInput}
								data-testid="password"
								name="password"
								type="password"
								label="Password"
							/>

							<Field
								component={CustomInput}
								data-testid="confirmPass"
								name="confirmPass"
								type="password"
								label="Confirm Password"
							/>

							<Box
								sx={{
									marginTop: '25px',
									paddingLeft: '15px',
								}}
							>
								<CustomSwitch 
									data-testid="rememberMe"
									name="rememberMe"
									label={
										<Typography
											component={'span'}
											sx={{
												fontSize: '14px',
												fontWeight: 300,
												lineHeight: '20px',
											}}
										>Remember me</Typography>
									}
									styles={{
										'& .MuiSwitch-switchBase': {
											'&.Mui-checked': {
												color:'#6FB7AC',
												'& + .MuiSwitch-track': {
													backgroundColor: '#6FB7AC',
												}
											},
										}}
									}
								/>
							</Box>

							<Box 
								sx={{
									marginTop: '25px',
									paddingLeft: '15px',
								}}
							>
								<CustomCheckBox 
									data-testid="subscribe"
									name="subscribe"
									styles={{
										color: '#6FB7AC',
										'&.Mui-checked': {
											color:'#6FB7AC',
										}}
									}
									label={
										<Typography
											component={'span'}
											sx={{
												fontSize: '14px',
												fontWeight: 300,
												lineHeight: '20px',
											}}
										>
											Let`s get personal! We`ll send you only the good 
											stuff: Exclusive early access to Sale, new arrivals 
											and promotions. No nasties.
										</Typography>
									}
								/>
							</Box>

							<p className={classes.policy}>By signing up you agree to
								<Link to="/info/terms-of-service"> Terms of Service </Link>
								and <Link to="/info/privacy-policy"> Privacy Policy </Link>
							</p>

							{serverResult && serverResult.error && (
								<Box className={classes.formStatusBlock}>
									<p className={classes.error}>{serverResult.error}</p>
								</Box>
							)}

							{serverResult && serverResult.success && (
								<Box className={classes.formStatusBlock}>
									<p className={classes.success}>{serverResult.success}</p>
								</Box>
							)}
							<Box sx={{ display: 'flex', justifyContent: 'center' }}>
								<Button
									type='submit'
									variant="contained"
									direction="form"
									disabled={
										!formikProps.isValid ||
										formikProps.isSubmitting
									}
								>
									sign up
								</Button>
							</Box>
						</Form>
					)
				}}
			</Formik>
		</Box>
	)
}

export default SignInForm