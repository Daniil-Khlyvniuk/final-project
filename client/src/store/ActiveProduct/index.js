
export { default as activeProductSelector } from './selectors'
export { default as activeProductOperations } from './operations'
import activeProductReducer from './activeProductSlice'

export default activeProductReducer