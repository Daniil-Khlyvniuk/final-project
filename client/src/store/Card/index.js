export { default as cardSelectors } from './selectors'
export { default as cardOperations } from './operations'
import * as cardActions from './cardSlice'

export default cardActions


