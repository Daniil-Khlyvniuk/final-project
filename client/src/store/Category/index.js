export { default as categorySelectors } from './selectors'
export { default as categoryOperations } from './operations'

import * as categoryActions from './categorySlice'
export default categoryActions