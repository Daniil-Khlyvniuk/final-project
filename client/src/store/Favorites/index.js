export { default as favoritesSelectors } from './selectors'
export { default as favoritesOperations } from './operations'

import * as favoritesActions from './favoritesSlice'
export default favoritesActions