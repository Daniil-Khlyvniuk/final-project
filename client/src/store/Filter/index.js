import reducer from './filterSlice'

export {default as filterSelectors} from './selectors'
export {default as filterOperations} from './operations'

export default reducer