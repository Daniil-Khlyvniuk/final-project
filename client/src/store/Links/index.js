export {default as linksSelectors} from './selectors'
export {default as linksOperations} from './operations'

import linksReducer from './linksSlice'
export default linksReducer