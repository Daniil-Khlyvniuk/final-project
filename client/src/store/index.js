import { configureStore } from '@reduxjs/toolkit'
import productsSlice from './Products/productsSlice'
import linksSlice from './Links/'
import categorySlice from './Category/categorySlice'
import sliderSlice from './Slider'
import modalSlice from './Modal/modalSlice'
import shoppingBagSlice from './ShoppingBag/shoppingBagSlice'
import productSlice from './Product/productSlice'
import filterSlice from './Filter'

import userSlice from './User'

const store = configureStore({
	reducer: {
		slides: sliderSlice,
		products: productsSlice,
		links: linksSlice,
		category: categorySlice,
		modal: modalSlice,
		shoppingBag: shoppingBagSlice,
		user: userSlice,
		product : productSlice,
		filter: filterSlice,

	},
	middleware: (getDefaultMiddleware) =>
		getDefaultMiddleware({
			serializableCheck: {
				// Ignore these action types
				ignoredActions: ['modal/modalToggle', 'user/setToken', 'user/fetchUser/pending','user/fetchUser/fulfilled','filter/setFiltersFromQueryString'],
			},
		}),
})

export default store
