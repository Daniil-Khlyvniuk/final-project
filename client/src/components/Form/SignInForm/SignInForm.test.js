import { render, screen, waitFor } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '../../../store'
import { BrowserRouter } from 'react-router-dom'
import userEvent from '@testing-library/user-event'
import React from 'react'
import SignInForm from './SignInForm'

describe('Smoke test SignInForm', () => {
	const setUp = () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<SignInForm />
				</BrowserRouter>
			</Provider>
		)
	}

	let container
	beforeEach(() => {
		container = setUp()
	})
	it('should render SignInForm', () => {
		expect(container)
	})
	it('SignInForm Snapshot test', () => {
		expect(container).toMatchSnapshot()
	})
})

describe("Testing formik on SignInForm", () => {
	it('should rendering and submitting a basic Formik form', () => {
		const handleSubmit = jest.fn()
		render(
			<Provider store={store}>
				<BrowserRouter>
					<SignInForm />
				</BrowserRouter>
			</Provider>
		)

		userEvent.type(screen.getByTestId('firstName'), 'Gogi')
		userEvent.type(screen.getByTestId('lastName'), 'GogiSurname')
		userEvent.type(screen.getByTestId('login'), 'nickname')
		userEvent.type(screen.getByTestId('email'), 'gogi123@gmail.com')
		userEvent.type(screen.getByTestId('password'), 'qwerty12345')
		userEvent.type(screen.getByTestId('confirmPass'), 'qwerty12345')

		userEvent.click(screen.getByTestId('submit', { type: /submit/i }))

		waitFor(() =>
			expect(handleSubmit).toHaveBeenCalledWith({
				firstname: 'Gogi',
				lastname: 'GogiSurname',
				login: 'nickname',
				email: 'gogi123@gmail.com',
				password: 'qwerty12345',
				confirmPass: 'qwerty12345'
			}),
		)
	})
})