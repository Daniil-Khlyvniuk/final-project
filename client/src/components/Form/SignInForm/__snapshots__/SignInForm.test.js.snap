// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke test SignInForm SignInForm Snapshot test 1`] = `undefined`;
